library/secrethub_read.py

```diff
@@ -68,7 +68,7 @@ def __init__(self):
                 'type': "str",
             },
         }
-        super(ReadModule, self).__init__(argument_spec)
+        super(ReadModule, self).__init__(argument_spec, supports_check_mode=True)
 
     def run(self):
         try:
```

Comment:
Should it perform a `read` when running in check mode (i.e. triggering an audit event and decrypting data)? In the other modules, your design looks like it only performs client calls when not in check mode.

Reply:
Also, what happens when the requested secret does not exist? Should it fail in check mode then?

Reply:
> Also, what happens when the requested secret does not exist? Should it fail in check mode then?

For example the `copy` module, fails in check mode when the source file does not exists: https://github.com/ansible/ansible/blob/v2.7.4/lib/ansible/modules/files/copy.py#L320

> Should it perform a read when running in check mode (i.e. triggering an audit event and decrypting data?

I think triggering audit events is a side-effect we should allow the module to have. This holds for running in check-mode, but they are also an exception to the idempotent behavior of the module: triggering the module multiple times will create multiple audit events. All other state will be untouched in subsequent runs.

However, in check-mode, it is not necessary to fetch the actual secret data. Checking that the secret version exists is enough. We could do so by doing an ls on the secret path (including version if given).
By doing so, the module does not do more than required and the audit events that are created when running in check-mode align better with the actions performed.

Reply:
Alright cool! So if I get this correctly the module will do the following in check mode:
- Fail if a prerequisite is not met
- Only check if a secret exists without reading it